File: counting/most-frequent-even-element.py
class Solution:
    def mostFrequentEven(self, nums: List[int]) -> int:
        count = {}
        for num in nums:
            if num%2 == 0:
                if num in count:
                    count[num] +=1
                else:
                    count[num] = 1
        maxKEY = 0
        maxValue = -1
        for key, value in count.items():
            if value > maxValue:
                maxValue = value
                maxKey = key
            elif maxValue == value and key<maxKey:
                maxValue = value
        return maxValue
        

File: array/most-frequent-even-element.cpp
class Solution {
public:
    int mostFrequentEven(vector<int>& nums) {
        int el;
        int prev;
        int count =0;
        for(int i = 0;i<nums.size();i++){
            if(count == 0 && nums.at(i)%2 == 0){
                count++;
                el = nums.at(i);
            }
            else if(nums.at(i)%2!=0){
                continue;
            }
            else if(nums.at(i) == el){
                count++;
            }
            else{
                count--;
                if(count == 0){
                    prev = el;
                }
            }
        }
        if(prev<el){
            return prev;
        }
        else{
            return el;
        }
        return -1;
    }
};